feat: 초대 응답 화면 UI 구현

jus0k opened the pull request:
### Short description

#12 의 하위 과제인 초대 응답 화면 UI 구현

### Proposed changes

- ReplyInvitation 페이지 컴포넌트 생성 및 라우트(/reply/:token) 연결
- Layout 화면 컴포넌트 추가
- 초대 응답 화면 프로필, 텍스트, 버튼 UI 추가
- 디자인 변경에 따른 Button, Header 공용 컴포넌트 스타일 변경
- VS Code ESlint 플러그인 업데이트에 따른 파일 저장시 포맷 적용 자동화 코드 업데이트

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionally)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Test
- [ ] Documentation
- [ ] Refactoring
- [x] Chore

### How to test (Optional)

- 제플린과 구현된 UI 비교

### Reference (Optional)

- Fixes #57

Commit:
feat: 초대 응답 화면 UI 구현 (#57)

- Reply 컴포넌트 생성
- /reply/:token 라우트 추가
- Layout 컴포넌트 추가
- 프로필 이미지 및 텍스트 추가
- 하단 full width 버튼 추가

Signed-off-by: jus0k <jutjd115@gmail.com>

Commit:
feat: 디자인 변경에 따른 Button, Header 공용 컴포넌트 스타일 변경

- Button 컴포넌트 텍스트 사이즈 button(14px)에서 caption(12px)로 변경 및 padding 추가
- Header 컴포넌트 하단 라인 색상 gray-300에서 gray-200으로 변경 및 padding 변경

Signed-off-by: jus0k <jutjd115@gmail.com>

chayeoi commented:
현재 Netlify의 Deploy Preview를 통해 배포된 앱에서 주소창에 `/invitation/reply/123`과 같이 루트 경로 이외의 URL로 직접 접속하면 "Page Not Found"가 반환되어 UI를 확인할 수 없는 이슈가 있습니다.
이 문제를 해결하기 위한 PR(#60)을 올려두었으니 해당 PR을 먼저 리뷰해주시고, 병합되고 난 후 현재 브랜치에 이 작업에 대한 sync를 맞춰주시면 그때 다시 리뷰 진행하도록 하겠습니다.
그 전까지 다른 리뷰어들이 해당 PR이 현재 작업 중인 상태임을 알 수 있도록 PR 타이틀에 접두사로 `[WIP]`를 추가해주시면 감사하겠습니다~

chayeoi commented on src/constants/RouteEnum.ts:
```diff
@@ -1,6 +1,7 @@
 enum RouteEnum {
   Home = '/',
   Invitation = '/invitation',
+  ReplyInvitation = '/invitation/reply/:rid',
```
초대 응답 화면이 초대 화면의 하위 하면이 아니므로 `/invitation/reply/:rid`와 같이 `/invitation`의 하위 경로로 작성하는 게 의미 상 맞지 않는 거 같습니다~ `/acceptance/acceptnaceId`와 같이 하는 건 어떨까요?
`rid`와 같이 축약해서 작성하면 나중에 알아보기 힘들기도 하고 'r'로 시작하는 다른 아이디 타입이 필요할 경우 중복되는 문제가 생길 것 같은데, `replyId`와 같이 축약하지 않은 형태로 작성하는 건 어떠신가요~?

jus0k replied:
초대에 응답하는 화면이라 하위 화면으로 보았는데 분리하는게 좋을 것 같네요! acceptance 는 수락의 의미라 응답에는 reply가 더 적절할 것 같아서 reply를 썼었는데요~ reply/:replyId로 사용하는건 어떨까요? 그리고 궁금한 점이 있는데, id같은 params의 경우 콜론을 보통 붙여서 썼었는데 붙이지 않아도 상관없는 것인지 궁금합니다!

chayeoi replied:
> 그리고 궁금한 점이 있는데, id같은 params의 경우 콜론을 보통 붙여서 썼었는데 붙이지 않아도 상관없는 것인지 궁금합니다!

콜론(`:`)을 작성해야 콜론 뒤에 작성한 이름의 변수로 URL 파라미터 값을 읽을 수 있습니다~

jus0k replied:
생각해보니 reply(answer, response 등등)를 하나의 독립된 라우터로 볼 수 있는가를 고민하고 초대의 하위 화면으로 작성을 한 것 같습니다! 이럴 때는 어떻게 라우트를 분리하는게 좋을 지 헷갈리네요

chayeoi replied:
> 초대에 응답하는 화면이라 하위 화면으로 보았는데 분리하는게 좋을 것 같네요! acceptance 는 수락의 의미라 응답에는 reply가 더 적절할 것 같아서 reply를 썼었는데요~ reply/:replyId로 사용하는건 어떨까요?

그렇네요~ 수락 뿐 아니라 거절도 가능하기 때문에 acceptance라는 표현이 적절하지 않은 것 같네요. 저는 다른 서비스에서 초대 기능을 구현할 때 초대 화면 pathname을 `/invitation`으로 지었었는데, 지금의 경우 초대 요청 화면에서 해당 경로가 쓰이고 있네요. 개인적인 생각에 reply는 뭔가 고객센터에서 Q&A에 대한 응답을 의미하는 듯한 뉘앙스가 있는 거 같아서 얘기드렸었습니다~ 우선 얘기주신대로 `/reply`로 한 다음, 추후에 프로필 설정과 초대하기 화면을 한 도메인의 하위 도메인으로 집어넣는 형태로 리팩토링하면서  `/invitation`으로 변경하면 좋을 듯합니다~

jus0k replied:
> > 초대에 응답하는 화면이라 하위 화면으로 보았는데 분리하는게 좋을 것 같네요! acceptance 는 수락의 의미라 응답에는 reply가 더 적절할 것 같아서 reply를 썼었는데요~ reply/:replyId로 사용하는건 어떨까요?
> 
> 그렇네요~ 수락 뿐 아니라 거절도 가능하기 때문에 acceptance라는 표현이 적절하지 않은 것 같네요. 저는 다른 서비스에서 초대 기능을 구현할 때 초대 화면 pathname을 `/invitation`으로 지었었는데, 지금의 경우 초대 요청 화면에서 해당 경로가 쓰이고 있네요. 개인적인 생각에 reply는 뭔가 고객센터에서 Q&A에 대한 응답을 의미하는 듯한 뉘앙스가 있는 거 같아서 얘기드렸었습니다~ 우선 얘기주신대로 `/reply`로 한 다음, 추후에 프로필 설정과 초대하기 화면을 한 도메인의 하위 도메인으로 집어넣는 형태로 리팩토링하면서 `/invitation`으로 변경하면 좋을 듯합니다~

넵 알겠습니다! 다른 단어도 찾아보니 answer, response 가 있었는데 마땅치 않은 것 같아 일단 임시로 사용한다 생각하고 `/reply`로 사용하겠습니다~

chayeoi replied:
추가로, `replyId`를 URL 파라미터로 받기보다는 `token`이라는 이름의 쿼리 파라미터로 받는 게 더 적절할 것 같습니다~ 현재 `replyId`라는 이름을 사용한 값은 `reply`에 대한 하위 리소스를 식별하는 역할이 아니라, 정해진 시간 내에 인증을 하는 용도로 사용되기 때문입니다~
[여기 링크](https://medium.com/@fullsour/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d) 내용을 참고해주시면 좋을 것 같습니다!

jus0k replied:
> 추가로, `replyId`를 URL 파라미터로 받기보다는 `token`이라는 이름의 쿼리 파라미터로 받는 게 더 적절할 것 같습니다~ 현재 `replyId`라는 이름을 사용한 값은 `reply`에 대한 하위 리소스를 식별하는 역할이 아니라, 정해진 시간 내에 인증을 하는 용도로 사용되기 때문입니다~
> [여기 링크](https://medium.com/@fullsour/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d) 내용을 참고해주시면 좋을 것 같습니다!

오 넵 알겠습니다~ 읽어보겠습니다!

jus0k replied:
> 추가로, `replyId`를 URL 파라미터로 받기보다는 `token`이라는 이름의 쿼리 파라미터로 받는 게 더 적절할 것 같습니다~ 현재 `replyId`라는 이름을 사용한 값은 `reply`에 대한 하위 리소스를 식별하는 역할이 아니라, 정해진 시간 내에 인증을 하는 용도로 사용되기 때문입니다~
> [여기 링크](https://medium.com/@fullsour/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d) 내용을 참고해주시면 좋을 것 같습니다!

오 넵 알겠습니다~ 읽어보겠습니다!

chayeoi replied:
> 생각해보니 reply(answer, response 등등)를 하나의 독립된 라우터로 볼 수 있는가를 고민하고 초대의 하위 화면으로 작성을 한 것 같습니다! 이럴 때는 어떻게 라우트를 분리하는게 좋을 지 헷갈리네요

예를 들어 게시글 목록 화면과 게시글 상세 화면의 관계는 하위 관계로 볼 수 있을 것 같습니다. 그러나 지금처럼 "초대 응답"과 "초대 요청"은 둘 중 하나가 다른 하나의 하위에 속한다고 보기는 어려운 것 같습니다~

jus0k replied:
> > 생각해보니 reply(answer, response 등등)를 하나의 독립된 라우터로 볼 수 있는가를 고민하고 초대의 하위 화면으로 작성을 한 것 같습니다! 이럴 때는 어떻게 라우트를 분리하는게 좋을 지 헷갈리네요
> 
> 예를 들어 게시글 목록 화면과 게시글 상세 화면의 관계는 하위 관계로 볼 수 있을 것 같습니다. 그러나 지금처럼 "초대 응답"과 "초대 요청"은 둘 중 하나가 다른 하나의 하위에 속한다고 보기는 어려운 것 같습니다~

아하 어떤 말씀이신지 알겠습니다!

chayeoi commented on src/views/App.tsx:
```diff
@@ -10,6 +10,7 @@ import { Spinner } from './components';
 import Home from './Home';
 import Invitation from './Invitation';
 import NotFound from './NotFound';
+import ReplyInvitation from './ReplyInvitation';
```
위에 남긴 코멘트에 따라 pathname을 `/accpentance/:acceptanceId`와 같이 바꾼다면, 그에 따라 `replyInvitation`에 해당하는 부분도 모두 이름을 변경해야 할 것 같습니다~

chayeoi commented on src/views/ReplyInvitation/index.ts:
```diff
@@ -0,0 +1,8 @@
+
+const ReplyInvitation = lazy(
+  (): Promise<{ default: React.ComponentType }> =>
+    import(/* webpackChunkName: 'ReplyInvitation', webpackPrefetch: true */ './ReplyInvitation'),
```
초대 응답 화면은 거의 사용되지 않는 화면이기 때문에, 전략적으로 이 화면은 prefetch를 적용하지 않는 게 좋을 것 같습니다~

jus0k replied:
아하 넵 알겠습니다~ 그럼 그냥 default export를 하면 되는건가요?

chayeoi replied:
아뇨! lazy 로드는 유지한 채 `webpackPrefetch: true` 주석만 제거해주시면 됩니다~

jus0k replied:
아하 알겠습니다!

chayeoi commented on src/views/ReplyInvitation/components/Layout.tsx:
```diff
@@ -0,0 +1,17 @@
+const s = {
+  wrapper: css`
+    width: 100%;
+    height: fit-content;
```
`height`에 `fit-content`를 사용하여야만 하는 이유가 있나요?
엣지 브라우저까지는 지원해도 좋을 것 같은데, 위 속성이 [Edge 브라우저에서 지원되지 않아서](https://developer.mozilla.org/ko/docs/Web/CSS/width) 다른 방법으로 대체 가능하면 변경해도 괜괜찮을 듯합니다~

jus0k replied:
`min-height` 프로퍼티를 사용해서 그랬는데 그냥 `min-height`을 없애고 `height: calc(100vh-100px);` 이렇게 변경하겠습니다!

chayeoi commented on src/views/components/Button/Button.tsx:
```diff
@@ -9,7 +9,8 @@ const Button: React.FC<ButtonProps> = (props) => <button css={s.button} {...prop
-    ${theme.typography.button};
+    ${theme.typography.caption};
+    color: ${theme.palette.gray[900]};
     padding: 0;
```
"닫기" 버튼 선택 영역이 너무 좁아서 모바일에서 클릭하기 쉽지 않을거 같습니다~ 제플린 시안대로 패딩을 추가하면 좋을 듯 싶습니다!

jus0k replied:
넵 반영하겠습니다~

chayeoi commented:
87bb6973e24ca8b97bd8ff961392160b9653ccbd 과 39795db15ea239f851449e29e90baf1313caa8ed 은 한 커밋으로 합쳐도 괜찮지 않을까 싶습니다~ 어떠신가요?

jus0k commented:
> [87bb697](https://github.com/mash-up-kr/luvket/commit/87bb6973e24ca8b97bd8ff961392160b9653ccbd) 과 [39795db](https://github.com/mash-up-kr/luvket/commit/39795db15ea239f851449e29e90baf1313caa8ed) 은 한 커밋으로 합쳐도 괜찮지 않을까 싶습니다~ 어떠신가요?

아하 넵 합치겠습니다!

Commit:
chore: vscode eslint 플러그인 업데이트에 따라 파일 저장시 포맷 적용 자동화 규칙 업데이트

VS Code ESlint 플러그인 업데이트에 따라 `editor.codeActionsOnSave` 규칙을 `source.fixAll.eslint: true`로 변경함

Signed-off-by: jus0k <jutjd115@gmail.com>

chayeoi commented on src/constants/RouteEnum.ts:
```diff
@@ -1,6 +1,7 @@
 enum RouteEnum {
   Home = '/',
   Invitation = '/invitation',
+  Reply = '/reply/:token',
```
`pathname`은 `/reply`로 하고, token은 `/reply?token={token}`와 같이 쿼리 파라미터의 형태로 받도록 하는게 좋은 것 같습니다~ 

> 추가로, replyId를 URL 파라미터로 받기보다는 token이라는 이름의 쿼리 파라미터로 받는 게 더 적절할 것 같습니다~ 현재 replyId라는 이름을 사용한 값은 reply에 대한 하위 리소스를 식별하는 역할이 아니라, 정해진 시간 내에 인증을 하는 용도로 사용되기 때문입니다~
여기 링크 내용을 참고해주시면 좋을 것 같습니다!

jus0k replied:
아하 그런 것이었군요!! 알겠습니다~
